crates/server/src/api/mod.rs

```diff
@@ -63,3 +79,72 @@ pub fn create_router(
                 ),
         )
 }
+
+/// An extractor that wraps the JSON extractor of Axum.
+///
+/// This extractor returns an API error on rejection.
+#[derive(FromRequest)]
+#[from_request(via(axum::Json), rejection(Error))]
+pub struct Json<T>(T);
```

Comment:
FYI: the reason these types were in the `v1` mod was that they return `v1` errors. In theory, a `v2` error might have a different format.

Reply:
Makes sense. I'll revert and have the debug API depend on v1 explicitly; it just requires making some private fields `pub(crate)`.